extensions/spark/kyuubi-spark-lineage/src/test/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParserHelperSuite.scala

```diff
@@ -690,6 +704,157 @@ class SparkSQLLineageParserHelperSuite extends KyuubiFunSuite
     }
   }
 
+  test("columns lineage extract - subquery sql") {
+    val ddls =
+      """
+        |create table table0(a int, b string, c string)
+        |create table table1(a int, b string, c string)
+        |""".stripMargin
+    ddls.split("\n").filter(_.nonEmpty).foreach(spark.sql(_).collect())
+    withTable("table0", "table1") { _ =>
+      val sql0 =
+        """
+          |select a as aa, bb, cc from (select b as bb, c as cc from table1) t0, table0
+          |""".stripMargin
+      val ret0 = exectractLineage(sql0)
+      assert(ret0 == Lineage(
+        List("default.table0", "default.table1"),
+        List(),
+        List(
+          ("aa", Set("default.table0.a")),
+          ("bb", Set("default.table1.b")),
+          ("cc", Set("default.table1.c")))))
+
+      val sql1 =
+        """
+          |select (select a from table1) as aa, b as bb from table1
+          |""".stripMargin
+      val ret1 = exectractLineage(sql1)
+      assert(ret1 == Lineage(
+        List("default.table1"),
+        List(),
+        List(
+          ("aa", Set("default.table1.a")),
+          ("bb", Set("default.table1.b")))))
+
+      val sql2 =
+        """
+          |select (select count(*) from table0) as aa, b as bb from table1
```

Comment:
should the input table include `table0` even we do not use it's columns ?  I'm not sure what's the behavior of other lineage extensions.

Reply:
Yes, that's what I was wondering about

Reply:
I add `table0.__aggregate__` to support aggregate expression.